File: CallingCFromJava/src/main/c/com/microej/example/java2c/NativeCCallExample.c
/*
 * C
 *
 * Copyright 2016-2019 MicroEJ Corp. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be found with this software.
 *
 */
#include <sni.h>

/**
* \brief
* \param	aValue
* \return	twice the input value, stored as a portable int
*/
jint Java_com_microej_example_java2c_NativeCCallExample_someCFunctionReturningTwiceAValue(jint aValue) {
	return aValue*2;
}

File: CallingCFromJava/src/main/java/com/microej/example/java2c/NativeCCallExample.java
/*
 *
 * Java
 *
 * Copyright 2016-2019 MicroEJ Corp. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be found with this software.
 *
 */
package com.microej.example.java2c;

public class NativeCCallExample {

	// No visible constructor for utility class
	private NativeCCallExample() {
	}

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		final int aValue = 120;
		System.out.println("Calling : someCFunctionReturningTwiceAValue(" + aValue +")");
		final int resultOfCallToSomeCFunction = someCFunctionReturningTwiceAValue(aValue);
		System.out.println("Result  : " + resultOfCallToSomeCFunction);
	}

	public static native int someCFunctionReturningTwiceAValue(int n);

}